USE_NFTW not available on build server?

## Steps to Reproduce the Problem

See https://build.libelektra.org/jenkins/blue/organizations/jenkins/libelektra/detail/PR-2135/26/pipeline

```sh

CYPTO        TESTS

==================



crypto_botan Results: 65 Tests done — 0 errors.

/home/jenkins/workspace/libelektra_PR-2135-KT7KVB3R7BMZPN4PF55YDXXT6WIH7TO44W3NZXZZ7PPZJAFP7MOA/tests/cframework/tests.c:542: error in clean_temp_home: Could not delete TMPHOME manually

--- running testmod_crypto_gcrypt ---



CYPTO        TESTS

==================



crypto_gcrypt Results: 65 Tests done — 0 errors.

/home/jenkins/workspace/libelektra_PR-2135-KT7KVB3R7BMZPN4PF55YDXXT6WIH7TO44W3NZXZZ7PPZJAFP7MOA/tests/cframework/tests.c:542: error in clean_temp_home: Could not delete TMPHOME manually

--- running testmod_crypto_openssl ---



CYPTO        TESTS

==================



crypto_openssl Results: 65 Tests done — 0 errors.

/home/jenkins/workspace/libelektra_PR-2135-KT7KVB3R7BMZPN4PF55YDXXT6WIH7TO44W3NZXZZ7PPZJAFP7MOA/tests/cframework/tests.c:542: error in clean_temp_home: Could not delete TMPHOME manually
```


## Expected Result

That Line 526 of tests.c is executed successfully.

## Actual Result

Line 542 is executed and fails.

## System Information

- Elektra Version: master

--- a/doc/news/_preparation_next_release.md
+++ b/doc/news/_preparation_next_release.md
@@ -511,6 +511,7 @@ Thanks to Daniel Bugl.
 - Setup mailing for jenkins *(Lukas Winkler)*
   - send mail to build@libelektra.org when `master` fails *(Lukas Winkler)*
   - parse change list into mail *(Lukas Winkler)*
+  - do not send mails if pipeline run was aborted *(Lukas Winkler)*
 
 ### Travis
 

--- a/doc/news/_preparation_next_release.md
+++ b/doc/news/_preparation_next_release.md
@@ -460,6 +460,7 @@ Thanks to Daniel Bugl.
 - The function `add_plugintest` now also supports setting environment variables for C/C++ based tests. *(René Schwaiger)*
 - The build system now automatically detects Homebrew’s OpenSSL version on macOS. *(René Schwaiger)*
 - We improved the automatic detection of Libgcrypt and OpenSSL. *(René Schwaiger)*
+- Resolved an issue where cmake did not properly set test feature macros to detect and use libc functionality. *(Lukas Winkler)*
 
 [Google Test]: https://github.com/google/googletest
 

--- a/scripts/jenkins/Jenkinsfile
+++ b/scripts/jenkins/Jenkinsfile
@@ -1435,3 +1435,9 @@ def detectInterruption(Closure c) {
     }
   }
 }
+
+class UserInterruptedException extends Exception {
+    UserInterruptedException(String message) {
+        super(message)
+    }
+}

--- a/scripts/jenkins/Jenkinsfile
+++ b/scripts/jenkins/Jenkinsfile
@@ -163,8 +163,12 @@ def main() {
 }
 
 try {
-  main()
-} catch(e) {
+  detectInterruption {
+    main()
+  }
+} catch(UserInterruptedException uie) {
+  println "Caught ${uie}"
+} catch(Exception e) {
   if (isMaster()) {
     // If master is failing we want to know ASAP so send a mail.
     def changes = currentBuild.changeSets.collect() {
@@ -184,11 +188,6 @@ Changes: ${RUN_CHANGES_DISPLAY_URL}
 ${changes}
 
 Logs: ${currentBuild.rawBuild.getLog(20).join('\n')}
-
----
-Debug Information for mail step:
-currentResult: ${currentBuild.currentResult}
-Exception: ${e}
 """
     mail subject: "Build ${JOB_NAME} failed",
          body: message,
@@ -1405,5 +1404,34 @@ def getElektraWritableFiles() {
 def ensureDirsExist(listOfDirs) {
   listOfDirs.each {
     sh "mkdir -p $it || /bin/true"
+  }
+}
+
+
+/* Detect if pipeline was aborted
+ *
+ * Depending on which part of the pipeline was interrupted a different Exception
+ * is thrown. This wrapper makes sure a UserInterruptedException is thrown
+ * regardless of pipeline state.
+ *
+ * see: https://issues.jenkins-ci.org/browse/JENKINS-34376
+ */
+def detectInterruption(Closure c) {
+  try {
+    c()
+  } catch (org.jenkinsci.plugins.workflow.steps.FlowInterruptedException fie) {
+    // this ambiguous condition means a user probably aborted
+    if (fie.causes.size() == 0) {
+        throw new UserInterruptedException(fie)
+    } else {
+        throw fie
+    }
+  } catch (hudson.AbortException ae) {
+    // this ambiguous condition means during a shell step, user probably aborted
+    if (ae.getMessage().contains('script returned exit code 143')) {
+        throw new UserInterruptedException(ae)
+    } else {
+        throw ae
+    }
   }
 }

--- a/tests/cframework/CMakeLists.txt
+++ b/tests/cframework/CMakeLists.txt
@@ -1,7 +1,15 @@
 include (CheckSymbolExists)
 include (LibAddMacros)
+include (CMakePushCheckState)
 
+# We need GNU_SOURCE and _XOEPN_SOURCE for full functinality
+set (REQUIRED_FEATURE_MACROS -D_GNU_SOURCE -D_XOPEN_SOURCE=500 -D_DARWIN_C_SOURCE)
+add_definitions (${REQUIRED_FEATURE_MACROS})
+
+cmake_push_check_state (RESET)
+set (CMAKE_REQUIRED_DEFINITIONS ${REQUIRED_FEATURE_MACROS})
 check_symbol_exists (nftw "ftw.h" HAVE_NFTW)
+cmake_pop_check_state ()
 
 if (HAVE_NFTW)
 	add_definitions (-DUSE_NFTW)

